src/networkWrapper/protocols/socketio.js

```diff
@@ -49,36 +43,85 @@ class SocketIO extends RTWrapper {
   }
 
   /**
-   * Registers a callback on a room. Once 1 message is received, fires the
+   * Registers a callback on an event. Once 1 message is received, fires the
    * callback and unregister it afterward.
    *
-   * @param {string} roomId
+   * @param {string} event
    * @param {function} callback
    */
-  once(roomId, callback) {
-    this.socket.once(roomId, callback);
+  addOnceListener(event, callback) {
+    return this.addListener(event, callback, true);
+  }
+
+  once(event, callback) {
+    return this.addOnceListener(event, callback);
+  }
+
+  prependOnceListener(event, callback) {
+    return this.prependListener(event, callback, true);
   }
 
   /**
-   * Registers a callback on a room.
+   * Registers a callback on an event.
    *
-   * @param {string} roomId
+   * @param {string} event
    * @param {function} callback
    */
-  on(roomId, callback) {
-    this.socket.on(roomId, callback);
+  addListener(event, callback, once = false) {
+    this._addEventWrapper(event, callback, once);
+    super.addListener(event, callback, once);
+
+    return this;
+  }
+
+  on(event, callback) {
+    return this.addListener(event, callback);
+  }
+
+  prependListener(event, callback, once = false) {
+    this._addEventWrapper(event, callback, once);
+    return this.prependListener(event, callback, once);
   }
 
   /**
-   * Unregisters a callback from a room.
+   * Unregisters a callback from an event.
    *
-   * @param {string} roomId
+   * @param {string} event
    * @param {function} callback
    */
-  off(roomId, callback) {
-    this.socket.off(roomId, callback);
+  removeListener(event, callback) {
+    if (this.eventsWrapper[event]) {
+      this.eventsWrapper[event].listeners.delete(callback);
+
+      if (this.eventsWrapper[event].listeners.size === 0) {
+        this.socket.off(event, this.eventsWrapper[event].wrapper);
+        delete this.eventsWrapper[event];
+      }
+
+      super.removeListener(event, callback);
+    }
+
+    return this;
   }
 
+  /**
+   * Unregisters all listeners either from an event, or from all events
+   *
+   * @param {string} [event]
+   */
+  removeAllListeners(event) {
+    if (event !== undefined && this.eventsWrapper[event] !== undefined) {
```

Comment:
if `event` is defined but `this.eventsWrapper[event]` is not, we will execute the `else` code, and so remove all listeners of all events ??